### ISSUE TITLE
Discovery / Registration only works when K8s pod already registered.

### ISSUE THREAD

**Issue by dzonekl** [codecentric/spring-boot-admin#1017]
Thanks for Spring Boot Admin!
(See also [1], which never received an answer).

I am using it with Spring Cloud Kubernetes, our k8s pods only get discovered when we start the Spring Boot Admin, after the service pods have been started.

It seems looking at InstanceDiscoveryListener, the discovery of clients, will happen based on events. Like ApplicationReadyEvent (When starting) and i.e. InstanceRegisteredEvent.

Is it correct to say, that Spring Boot Admin, will not try to discover periodically? 

I would like to make sure, that instances are registered when they are started after spring boot admin, was started. (The order in which k8s pods are started is arbitrary/hard to control, something in general we don't want to do).

Thank You! Christophe

[1] https://stackoverflow.com/questions/52763066/discovery-registration-only-works-when-k8s-pod-already-registered

[2] versions
    springBootAdminVersion = '2.0.1'
    springCloudVersion = 'Finchley.RELEASE'
    springCloudK8s = '0.3.0.RELEASE'

**Comment by joshiste** [comment]
This is due to a bug in spring-cloud-kubernetes. see #643 for more information.

**Comment by dzonekl** [comment]
I investigated further and stumbled upon the K8sCatalogWatcher, which is disabled by default. 
So this is also the bug you referred to. It is unfortunately not documented, so going through the code, I figured it out: 

So enabling the catalog watcher, and also enabling scheduling solved the issue: 

1. SpringBootApplication add: 
`@EnableScheduling`

2. In properties (yaml) add: 
```
spring:
  cloud:
    kubernetes:
        catalog-services-watch:
          enabled: true
          catalogServicesWatchDelay: 10000  (Default is 30 secs).
```
Now spring boot admin, will get a heartbeat.
